### PR TITLE
fix(combobox, combobox-item): Toggling 'selected' on combobox-item should update the combobox parent. #2102

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #2102

## Summary

fix(combobox, combobox-item): Toggling 'selected' on combobox-item should update the combobox parent. #2102

### CODE REVIEW

**Comment by driskull** [comment]
Still active

**Comment by driskull** [comment]
@jcfranco can you review? The PR is stale :)

File: src/components/calcite-combobox/calcite-combobox.tsx [hunk 1]
```diff
@@ -137,7 +137,7 @@ export class CalciteCombobox {
 
   @Listen("calciteComboboxItemChange")
   calciteComboboxItemChangeHandler(event: CustomEvent<HTMLCalciteComboboxItemElement>): void {
-    this.toggleSelection(event.detail);
+    this.toggleSelection(event.detail, event.detail.selected);
```
Comment (jcfranco) [inline src/components/calcite-combobox/calcite-combobox.tsx:140]: I think we should drop `selected` and go through the event's target.

File: src/components/calcite-combobox/calcite-combobox.e2e.ts [hunk 2]
```diff
@@ -311,11 +311,11 @@ describe("calcite-combobox", () => {
 
       await input.press("Enter");
       expect(await item1.getProperty("selected")).toBe(true);
-      expect(eventSpy).toHaveReceivedEventTimes(1);
+      expect(eventSpy).toHaveReceivedEventTimes(2);
 
       await input.press("Enter");
       expect(await item1.getProperty("selected")).toBe(false);
-      expect(eventSpy).toHaveReceivedEventTimes(2);
+      expect(eventSpy).toHaveReceivedEventTimes(4);
```
Comment (jcfranco) [inline src/components/calcite-combobox/calcite-combobox.e2e.ts:318]: The lookup event should only occur once on `Enter`, shouldn't it?

Comment (driskull) [reply src/components/calcite-combobox/calcite-combobox.e2e.ts:318]: Well, if there's already one selected then that one becoming unselected also fires an event.

Comment (jcfranco) [reply src/components/calcite-combobox/calcite-combobox.e2e.ts:318]: Technically, yes, but should it? It's one operation firing 2 change events. I think the pick/value list tackled something similar some time ago by debouncing the emitter.

Comment (driskull) [reply src/components/calcite-combobox/calcite-combobox.e2e.ts:318]: Gotcha. yeah i'll update that 👍

**Review by jcfranco — APPROVED** [review]
Awesome! 💥

**Pull request merged**